examples/Example/src/prot_snap/index.tsx

```diff
@@ -0,0 +1,347 @@
+import { useAnimatedGesture } from '../useAnimatedGesture';
+
+const filters = ['red', 'green', 'blue', 'yellow', 'orange', 'cyan'];
+const MAX_VIDEO_DURATION = 60000;
+
+export default function Home() {
+  const filter = useSharedValue(0);
+  const filterOffset = useSharedValue(0);
+  const zoom = useSharedValue(1);
+  const [scale, setScale] = useState(1);
+  const [selectedFilter, setSelectedFilter] = useState(0);
+  const [recording, setRecording] = useState(false);
+  const [remainingTime, setRemainingTime] = useState(MAX_VIDEO_DURATION);
+  const [recordingInterval, setRecordingInterval] = useState(-1);
+
+  const filtersPanGesture = Gesture.pan()
+    .setOnUpdate((e) => {
+      'worklet';
+      filter.value =
+        filter.value + (filterOffset.value - e.translationX) * 0.01;
+      filterOffset.value = e.translationX;
+
+      runOnJS(updateSelectedFilter)();
+    })
+    .setOnEnd(() => {
+      'worklet';
+      filterOffset.value = 0;
+      runOnJS(stopFilterScroll)();
+    });
+
+  const buttonTapGesture = Gesture.tap()
+    .setMaxDistance(3)
+    .setOnEnd((e, success) => {
+      'worklet';
+      if (success) runOnJS(takePhoto)();
+    });
+
+  const buttonDoubleTapGesture = Gesture.tap()
+    .setTapCount(2)
+    .setMaxDistance(3)
+    .setPriority(1)
+    .setOnEnd((e, success) => {
+      'worklet';
+      if (success) {
+        runOnJS(takeSeries)();
+      }
+    });
+
+  const buttonPanGesture = Gesture.pan()
+    .addSimultaneousGesture(filtersPanGesture)
+    .setOnUpdate((e) => {
+      'worklet';
+      if (recording) {
+        if (e.velocityY < 0) {
+          zoom.value = zoom.value * 1.05;
+        } else if (e.velocityY > 0) {
+          zoom.value = zoom.value * 0.95;
+        }
+      }
+    })
+    .setOnEnd((e) => {
+      'worklet';
+      if (recording) {
+        runOnJS(finishRecording)();
+      }
+    });
+
+  const buttonLongPressGesture = Gesture.longPress().setOnStart(() => {
+    'worklet';
+    runOnJS(startRecording)();
+  });
+
+  const previewPinchGesture = Gesture.pinch()
+    .setOnStart(() => {
+      'worklet';
+      runOnJS(setScale)(zoom.value);
+    })
+    .setOnUpdate((e) => {
+      'worklet';
+      zoom.value = scale * e.scale;
+    });
+
+  const panGestureHandler = useAnimatedGesture(filtersPanGesture);
+
+  const buttonGestureHandler = useAnimatedGesture(
+    buttonPanGesture
+      .after(buttonLongPressGesture)
+      .simultaneousWith(buttonTapGesture)
+      .simultaneousWith(buttonDoubleTapGesture)
+  );
+
+  const scaleGestureHandler = useAnimatedGesture(previewPinchGesture);
+
+  if (remainingTime <= 0) {
+    finishRecording();
+  }
+
+  function stopFilterScroll() {
+    filter.value = withTiming(updateSelectedFilter(), { duration: 200 });
+  }
+
+  function updateSelectedFilter() {
+    let targetFilter = Math.round(filter.value);
+    if (targetFilter < 0) targetFilter = 0;
+    if (targetFilter >= filters.length) targetFilter = filters.length - 1;
```

Comment:
Please use braces around `if` body, even if it's oneliner.
This is basically a clamp function and is commonly written in form of:
```ts
Math.min(Math.max(targetFilter, 0), filters.length - 1)
```
This will also allow for inlining `Math.round()` call and changing `let` to `const` (although you may introduce another variable for it to keep readability).

Reply:
Fixed in 935fdda5be6dcc25dda6fab0f25c373087593ad7